src/ORM/Association/BelongsToMany.php

```diff
@@ -387,7 +387,9 @@ protected function _appendNotMatching($query, $options)
             ->where($options['conditions'])
             ->andWhere($this->junctionConditions());
 
-        $subquery = $options['queryBuilder']($subquery);
+        if (!empty($options['queryBuilder'])) {
```

Comment:
Should that be

    if ($options['queryBuilder']) {

?
Because if this cannot be undefined this should be used instead of cloaking.